java/vector/src/main/java/org/apache/arrow/vector/UInt2Vector.java

```diff
@@ -308,6 +308,11 @@ public TransferPair makeTransferPair(ValueVector to) {
     return new TransferImpl((UInt2Vector) to);
   }
 
+  @Override
+  public void setEncodedValue(int index, int value) {
+    this.setSafe(index, value);
```

Comment:
please add asserts or precondition.checkArguments to ensure the value doesn't overflow for all types with smaller bitwidths